### ISSUE TITLE
modelsummary can't be used without loading modelsummary unless gof_map is specified

### ISSUE THREAD

**Issue by NickCH-K** [vincentarelbundock/modelsummary#236]
Maybe there's something weird about my install, but I can't use `modelsummary()` unless I load the **modelsummary** package.

```r
m <- lm(mpg ~ hp, data = mtcars)
modelsummary::modelsummary(m)
```
 Produces the error:

```
Error in as.environment("package:modelsummary") : 
  no item called "package:modelsummary" on the search list
```

This is specifically tied to the `gof_map` option:

```r
modelsummary::modelsummary(m, gof_map = modelsummary::gof_map)
```

works fine.

This appears to be due to line 398 in `modelsummary.R`:

```r
gm_list <- get("gof_map", as.environment("package:modelsummary"))
```

If I change this line to 
```r
gm_list <- modelsummary::gof_map
```

the problem goes away.

I didn't make a PR so as to not have two PRs going at once. But unless there's some reason that `get()` is preferred, I'll make a PR as soon as my last one is done.

**Comment by vincentarelbundock** [comment]
Oh, interesting.

I'm not 100% sure why I did that, but this is not the kind of code I usually write, so I'm guessing this was an intentional decision. 

My hunch is that this may have had to do with object name conflict: (1) `modelsummary` calls `is.null` on the user-supplied `gof_map`, and we don't want a name collision, and (2) we can't call `modelsummary::gof_map` because CRAN will bounce packages that refer to their own namespace with `::`. So I had to find an alternative to the obvious `modelsummary::gof_map`.

Does something like this work?

```r
get("gof_map", envir = loadNamespace("modelsummary"))
```
